src/sentry/utils/snuba.py

```diff
@@ -911,7 +945,12 @@ def constrain_column_to_dataset(col, dataset, value=None):
         return DATASETS[dataset][col]
     if col in DATASET_FIELDS[dataset]:
         return col
-    return u"tags[{}]".format(col)
+
+    if dataset == Dataset.Events:
+        return u"tags[{}]".format(col)
+    else:
+        # If we're not using the events dataset, we need to prefix the tags.
+        return u"events.tags[{}]".format(col)
```

Comment:
The groups dataset is the only one that needs prefixing, you should make this the special case in my opinion.